packages/admin/admin/src/mui/menu/Menu.styles.ts

```diff
@@ -1,58 +1,66 @@
-import { Theme } from "@mui/material";
-import { createStyles } from "@mui/styles";
+import { CSSObject, Drawer as MuiDrawer, DrawerProps, Theme } from "@mui/material";
+import { styled } from "@mui/material/styles";
+import { createStyles, StyledComponent } from "@mui/styles";
+import { MUIStyledCommonProps } from "@mui/system";
+import * as React from "react";
 
-import { MenuProps } from "./Menu";
+import { MasterLayoutContext } from "../MasterLayoutContext";
+import { DEFAULT_DRAWER_WIDTH, DEFAULT_DRAWER_WIDTH_COLLAPSED, MenuProps } from "./Menu";
 
-export type MenuClassKey = "drawer" | "permanent" | "temporary" | "open" | "closed";
+const openedMixin = (theme: Theme, drawerWidth?: number): CSSObject => ({
+    width: drawerWidth ?? DEFAULT_DRAWER_WIDTH,
+    transition: theme.transitions.create("width", {
+        easing: theme.transitions.easing.sharp,
+        duration: theme.transitions.duration.enteringScreen,
+    }),
+    overflowX: "hidden",
+});
+const closedMixin = (theme: Theme, drawerVariant: DrawerProps["variant"], drawerWidth?: number, drawerWidthCollapsed?: number): CSSObject => ({
+    width: drawerVariant === "temporary" ? drawerWidth ?? DEFAULT_DRAWER_WIDTH : drawerWidthCollapsed ?? DEFAULT_DRAWER_WIDTH_COLLAPSED,
+    transition: theme.transitions.create("width", {
+        easing: theme.transitions.easing.sharp,
+        duration: theme.transitions.duration.leavingScreen,
+    }),
+    overflowX: "hidden",
+});
+export const Drawer: StyledComponent<DrawerProps & MUIStyledCommonProps<Theme> & Pick<MenuProps, "drawerWidth" | "drawerWidthCollapsed">> = styled(
+    MuiDrawer,
+    { shouldForwardProp: (prop) => prop !== "drawerWidth" && prop !== "drawerWidthCollapsed" },
+)<DrawerProps & MUIStyledCommonProps<Theme> & Pick<MenuProps, "drawerWidth" | "drawerWidthCollapsed">>(
+    ({ theme, open, variant, drawerWidth, drawerWidthCollapsed }) => {
+        const { headerHeight } = React.useContext(MasterLayoutContext);
 
-export const styles = (theme: Theme) =>
-    createStyles<MenuClassKey, MenuProps>({
-        drawer: {
-            "& [class*='MuiDrawer-paper']": {
-                backgroundColor: "#fff",
-            },
-            "& [class*='MuiPaper-root']": {
-                flexGrow: 1,
-                overflowX: "hidden",
+        return {
+            ...(variant === "permanent" && {
+                backgroundColor: theme.palette.common.white,
+                flexShrink: 0,
+                whiteSpace: "nowrap",
+                boxSizing: "border-box",
+                ...(open ? openedMixin(theme, drawerWidth) : closedMixin(theme, variant, drawerWidth, drawerWidthCollapsed)),
+            }),
+            "& .MuiDrawer-paper": {
```

Comment:
```suggestion
            [`& .${drawerClasses.paper}`]: {
```

(Can also be done in follow-up PR)

Reply:
Fixed here: https://github.com/vivid-planet/comet/commit/7a95bdc96490482eba0ab05c7febef5df608656d